### PR TITLE
Rework the existing controller legacy tests

### PR DESCRIPTION
**Pull request by wilsonge**
Reworks the JControllerLegacy tests. This uses the @testdox tags that we are rolling out across Joomla and also breaks up the 11 parts of the setRedirect into separate tests. Finally I've added a test for injecting the controller name in the controller

### CODE REVIEW

**Comment by zero-24** [comment]
@wilsonge do we need to test here or only need travis pass the tests?

**Comment by wilsonge** [comment]
Just travis - I'm not touching core at all

**Comment by SniperSister** [comment]
A few comments:

* personally I would prefer "speaking" test names instead of names like test__construct()
* testCreateFileName() could be simplified by using a data provider
* setRedirect() could also be simplified by using a data provider, but using separate tests is fine for me too

All in all: good job! :)

**Comment by zero-24** [comment]
Thanks @wilsonge moving to RTC since Travis is happy with you here :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6441">issues.joomla.org/joomla-cms/6441</a>.</sub>

File: tests/unit/suites/libraries/legacy/controller/JControllerLegacyTest.php [hunk 1]
```diff
@@ -142,14 +136,31 @@ public function test__construct()
 	}
 
 	/**
-	 * Test JControllerLegacy::addPath().
+	 * @testdox  Ensure the constructor correctly initialises the class variables
```
Comment (mbabker) [inline tests/unit/suites/libraries/legacy/controller/JControllerLegacyTest.php]: Like the method names, the `@testdox` tags should be unique per test too.  You're obviously doing something with this test against the constructor compared to the previous test.

**Comment by zero-24** [comment]
moving back to pending as we have feedback by @SniperSister and @mbabker <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6441">issues.joomla.org/joomla-cms/6441</a>.</sub>

**Comment by wilsonge** [comment]
@SniperSister the setRedirect method is pretty hard to do via a data provider because half the tests also involve the setMessage method and how they override each other.

File: tests/unit/suites/libraries/legacy/controller/JControllerLegacyTest.php (hunk 1 above)
Comment (wilsonge) [inline tests/unit/suites/libraries/legacy/controller/JControllerLegacyTest.php]: Oops my bad. Fixed

**Comment by SniperSister** [comment]
@wilsonge oh yeah, true. Well, then let's just keep them as separate tests.

**Comment by mbabker** [comment]
Merged via https://github.com/joomla/joomla-cms/commit/a9a77da814b3512504592401d930811bf835f288

**Pull request closed without merging**